chirp/drivers/btech.py

```diff
@@ -923,8 +930,92 @@ def get_memory(self, number):
                                   SPMUTE_LIST[_mem.spmute]))
         mem.extra.append(spmute)
 
+        immutable = []
+
+        if self._gmrs:
+            if self.MODEL == "GMRS-50X1":
+                if mem.freq in GMRS_FREQS:
+                    if mem.number >= 1 and mem.number <= 30:
+                        GMRS_FREQ = GMRS_FREQS[mem.number - 1]
+                        mem.freq = GMRS_FREQ
+                        immutable = ["empty", "freq"]
+                    if mem.number >= 1 and mem.number <= 7:
+                        mem.duplex = ''
+                        mem.offset = 0
+                        mem.power = POWER_LEVELS[2]
+                        immutable += ["duplex", "offset", "power"]
+                    elif mem.number >= 8 and mem.number <= 14:
+                        mem.duplex = 'off'
+                        mem.offset = 0
+                        immutable += ["duplex", "offset"]
+                    elif mem.number >= 15 and mem.number <= 22:
+                        mem.duplex = ''
+                        mem.offset = 0
+                        immutable += ["duplex", "offset"]
+                    elif mem.number >= 23 and mem.number <= 30:
+                        mem.duplex = '+'
+                        mem.offset = 5000000
+                        immutable += ["duplex", "offset"]
+                else:
```

Comment:
Er, this makes it look like the channels aren't all immutable.. can you store non-sanctioned channels in it? If so, I think it needs the policy override so that we don't apply the immutable rules of the channel you're writing to to the thing you're trying to store. The policy check lets you basically say "just because something is immutable in channel X right now doesn't mean it will be immutable if you put something different into it". By my read of this section from L936 to L964, that needed override applies to the 50X1.

Reply:
Since immutable = ["empty", "freq"] at L941, immutable becomes ["empty", "freq", "duplex", "offset"] at line 958. That should prevent it from being deleted and from changing the frequency. Right?

Reply:
Ah, because there's no `elif` at L942... okay. So only memories 1-30 are fixed, but 0, and 30-255 are allowed to be anything, but without transmit ability regardless of what they are?

Reply:
Ah, because there's no `elif` at L942... okay. So only memories 1-30 are fixed, but 0, and 30-255 are allowed to be anything, but without transmit ability regardless of what they are?

Reply:
Correct.

Reply:
Okay, so the immutable-ness of 1-30 is always sticky, but the immutable-ness of 0,30-255 is not (although it will be overridden anyway). So I think the policy check should only set `immutable=[]` if the number is not 1-30, and then make validate return a warning for the "duplex gonna be off" thing. This might be too obscure for the tests, so if it seems like it's all right and still not passing after that, push it up again and I'll have a look at it.

Reply:
I'm not 100% sure what you mean by "the policy check should only set immutable=[] if the number is not 1-30." Do you mean something like this?

    def check_set_memory_immutable_policy(self, existing, new):
        if not (mem.number >= 1 and mem.number <= 30):
            existing.immutable = []
            super().check_set_memory_immutable_policy(existing, new)

Other than having done this part, everything else seems to work OK and passes the tests.

Reply:
Gotcha...

Reply:
It tells me "NameError: name 'mem' is not defined. How do I pass the memory number to this?

Reply:
oh, I didn't read close enough. You want to check `new.number`. Both `new` and `existing` are memories.

Reply:
I think that got it. I forgot to fetch and rebase before I sent it again. I can to that in a little bit if you need me to.

Reply:
I rebased it, should be okay, thanks!